Mathlib/Algebra/DirectSum/Internal.lean

```diff
@@ -428,3 +438,93 @@ theorem SetLike.Homogeneous.smul [CommSemiring S] [Semiring R] [Algebra S R] {A
   ⟨i, Submodule.smul_mem _ _ hi⟩
 
 end HomogeneousElement
+
+/-! ### Gradings by canonically linearly ordered additive monoids -/
+
+section LinearOrderedAddCommMonoid
+
+variable [LinearOrderedAddCommMonoid ι] [DecidableEq ι]
+
+section Semiring
+
+variable [Semiring R] [SetLike σ R] [AddSubmonoidClass σ R]
+variable {A : ι → σ} [SetLike.GradedMonoid A]
+
+theorem mul_apply_eq_zero {r r' : ⨁ i, A i} {m n : ι}
+    (hr : ∀ i < m, r i = 0) (hr' : ∀ i < n, r' i = 0) ⦃k : ι⦄ (hk : k < m + n) :
+    (r * r') k = 0 := by
```

Comment:
@adomani, do we have this anywhere for `AddMonoidAlgebra`?

Reply:
Are you thinking of something like [this](https://github.com/leanprover-community/mathlib4/pull/6633#discussion_r1297225421)?

Reply:
Ah, I think that I found what I was looking for: [this comment of yours](https://github.com/leanprover-community/mathlib4/pull/6627#discussion_r1929335841).  I have not PRed this lemma, though.